traitlets/config/configurable.py

```diff
@@ -203,8 +210,13 @@ def _config_changed(self, change):
         section_names = self.section_names()
         self._load_config(change.new, traits=traits, section_names=section_names)
 
-    def update_config(self, config):
-        """Update config and load the new values"""
+    def update_config(self, config, skip_env=False):
```

Comment:
I always try to have boolean defaulting to `False`.
That's the default for traits, and many other programming domains.
Also it allows to work as planned if `None` is given.

Now why should it be reversed?
If someone puts in his code the `metadata['envvar']='SOMEVAR'`, why would he not want to use this, by default?
If he didn't like it, he wouldn't put the env-var there, it in the first place.  
He could have used dynamic-defaults
That's the essence of this PR, no?